src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -1139,7 +1164,359 @@ static void Main()
         }
 
         [Fact]
-        public void WithUserDefinedEqualityOperator()
+        public void EqualityComparison_InCSharp7_2()
+        {
+            string template = @"
+TYPE1 MyType
+{
+    static void Main()
+    {
+        TYPE2 x = VALUE;
+
+        if ((x == default) != EQUAL) throw null;
+        if ((default == x) != EQUAL) throw null;
+
+        if ((x != default) == EQUAL) throw null;
+        if ((default != x) == EQUAL) throw null;
+
+        if ((x == default(TYPE2)) != EQUAL) throw null;
+        if ((x != default(TYPE2)) == EQUAL) throw null;
+
+        System.Console.Write(""Done"");
+    }
+}
+";
+            validate("class", "int", "0", "true", "System.Int32");
+            validate("class", "int", "1", "false", "System.Int32");
+            validate("class", "string", "null", "true", "System.String");
+            validate("class", "string", "\"\"", "false", "System.String");
+
+            validate("class", "MyType", "null", "true", "System.Object",
+                // (8,14): error CS8320: Feature 'default equality' is not available in C# 7.2. Please use language version 7.3 or greater.
+                //         if ((x == default) != true) throw null;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_2, "x == default").WithArguments("default equality", "7.3").WithLocation(8, 14),
+                // (9,14): error CS8320: Feature 'default equality' is not available in C# 7.2. Please use language version 7.3 or greater.
+                //         if ((default == x) != true) throw null;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_2, "default == x").WithArguments("default equality", "7.3").WithLocation(9, 14),
+                // (11,14): error CS8320: Feature 'default equality' is not available in C# 7.2. Please use language version 7.3 or greater.
+                //         if ((x != default) == true) throw null;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_2, "x != default").WithArguments("default equality", "7.3").WithLocation(11, 14),
+                // (12,14): error CS8320: Feature 'default equality' is not available in C# 7.2. Please use language version 7.3 or greater.
+                //         if ((default != x) == true) throw null;
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion7_2, "default != x").WithArguments("default equality", "7.3").WithLocation(12, 14)
+                );
+
+            validate("struct", "MyType", "new MyType()", "true", "System.Object",
+                // (8,14): error CS8320: Feature 'default equality' is not available in C# 7.2. Please use language version 7.3 or greater.
```

Comment:
>error CS8320: Feature 'default equality' is not available in C# 7.2. Please use language version 7.3 or greater. [](start = 27, length = 112)

This error is misleading. Since the equality cannot be applied, we should not be offering to change language version.